packages/flutter_hooks/lib/src/platform_brightness.dart

```diff
@@ -0,0 +1,60 @@
+part of 'hooks.dart';
+
+/// A callback triggered when the platform brightness changes.
+typedef BrightnessCallback = FutureOr<void> Function(
+  Brightness? brightness,
+);
+
+/// Returns the current Platform Brightness value and rebuilds the widget when it changes.
+Brightness? usePlatformBrightness() {
+  return use(const _PlatformBrightnessHook(rebuildOnChange: true));
+}
+
+useOnPlatformBrightnessChange(BrightnessCallback? onBrightnessChange) {
+  return use(_PlatformBrightnessHook(onBrightnessChange: onBrightnessChange));
+}
+
+class _PlatformBrightnessHook extends Hook<Brightness?> {
+  const _PlatformBrightnessHook({
+    this.rebuildOnChange = false,
+    this.onBrightnessChange,
+  }) : super();
+
+  final bool rebuildOnChange;
+  final BrightnessCallback? onBrightnessChange;
+
+  @override
+  _PlatformBrightnessState createState() => _PlatformBrightnessState();
+}
+
+class _PlatformBrightnessState
+    extends HookState<Brightness?, _PlatformBrightnessHook>
+    with WidgetsBindingObserver {
+  Brightness? _brightness;
```

Comment:
Should be non-nullable, using `late`

platformBrightness is never null, and WidgetsBinding.instance should never be null either